### PR TITLE
chore(clients/go): skip license test for all mocks

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Skip license test for all Go mocks.

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/go/test/license_test.go [hunk 1]
```diff
@@ -40,8 +40,7 @@ const license = `// Copyright © 2018 Camunda Services GmbH (info@camunda.com)
 
 var skip = map[string]bool{
 	// These files are generated.
```
Comment (MiguelPires) [inline clients/go/test/license_test.go:42]: What if we turn this into a "skipList" with the prefixes of files/dirs to ignore? Then we wouldn't need to add a new if clause for every directory we want to ignore and we're not actually using this map as a map. Something like:

```suggestion
var skipList = []string{
  // These files are generated.
  "../pkg/pb/gateway.pb.go",
  "../internal/mock_pb/",
  "../vendor/",
}

func TestLicense(t *testing.T) {
	err := filepath.Walk("..", func(path string, fi os.FileInfo, err error) error {
    for _, skip := range skipList {
      if strings.HasPrefix(path, skip) {
        return nil
      }
    }
```

**Review by miguelpires — CHANGES_REQUESTED** [review]

File: clients/go/test/license_test.go (hunk 1 above)
Comment (npepinpe) [reply clients/go/test/license_test.go:42]: Sounds good :+1:

**Review by miguelpires — APPROVED** [review]
bors r+

**Comment by npepinpe** [comment]
bors retry

**Pull request merged**